Python/Product/Analysis/Interpreter/PythonInterpreterFactoryExtensions.cs

```diff
@@ -52,6 +54,19 @@ public static class PythonInterpreterFactoryExtensions {
         /// </summary>
         /// <returns>The names of the modules that were found.</returns>
         public static async Task<HashSet<string>> FindModulesAsync(this IPythonInterpreterFactory factory, params string[] moduleNames) {
+            var withPackages = factory as IPackageManager;
+            if (withPackages != null) {
+                var res = new HashSet<string>();
+                foreach (var m in moduleNames) {
+                    if ((await withPackages.GetInstalledPackageAsync(new PackageSpec(m), CancellationToken.None)).IsValid) {
+                        res.Add(m);
+                    }
+                }
+                if (res.Count == moduleNames.Length) {
```

Comment:
Sure. That sounds like a good way to avoid bugs 

---
In reply to: [75513879](https://github.com/Microsoft/PTVS/pull/1536#discussion_r75513879) [](ancestors = 75513879)